packages/component/src/providers/CustomElements/customElements/CodeBlock.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable class-methods-use-this */
+import { hooks } from 'botframework-webchat-api';
+import { ReactNode, useMemo } from 'react';
+import { useRefFrom } from 'use-ref-from';
+
+import { useStyleSet } from '../../../hooks';
+import { defaultHighlightCode, HighlightCodeFn } from '../../../hooks/internal/codeHighlighter';
+import { parseDocumentFragmentFromString, useCodeHighlighter } from '../../../internal';
+import { createElementRegistryWithUpdater } from '../private/useUpdater';
+
+const { useStyleOptions, useLocalizer } = hooks;
+
+class CodeBlock extends HTMLElement {
+  static observedAttributes = ['theme', 'language'];
+
+  connected = false;
```

Comment:
Should we mark this is private?

```suggestion
  #connected = false;
```